src/transformStateWithClones.js

```diff
@@ -7,7 +7,24 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
-}
+  const result = [];
+
+  let transClone = structuredClone(state);
+
+  for (const action of actions) {
+    if (action.type === 'removeProperties') {
+      for (const key of action.keysToRemove) {
+        delete transClone[key];
```

Comment:
When deleting properties, ensure that the keys in `action.keysToRemove` exist in `transClone` to avoid unnecessary operations.